xdsl/dialects/smt.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import annotations
+
+from collections.abc import Sequence
+from typing import TypeAlias
+
+from xdsl.dialects.builtin import BoolAttr, IntegerAttr
+from xdsl.ir import Dialect, ParametrizedAttribute, SSAValue, TypeAttribute
+from xdsl.irdl import (
+    AtLeast,
+    IRDLOperation,
+    RangeOf,
+    base,
+    irdl_attr_definition,
+    irdl_op_definition,
+    prop_def,
+    result_def,
+    traits_def,
+    var_operand_def,
+)
+from xdsl.traits import ConstantLike, Pure
+
+
+@irdl_attr_definition
+class BoolType(ParametrizedAttribute, TypeAttribute):
+    """A boolean."""
+
+    name = "smt.bool"
+
+
+NonFuncSMTType: TypeAlias = BoolType
+
+
+@irdl_op_definition
+class ConstantBoolOp(IRDLOperation):
+    """
+    This operation represents a constant boolean value. The semantics are
+    equivalent to the ‘true’ and ‘false’ keywords in the Core theory of the
+    SMT-LIB Standard 2.7.
+    """
+
+    name = "smt.constant"
+
+    value_attr = prop_def(BoolAttr, prop_name="value")
+    result = result_def(BoolType())
+
+    traits = traits_def(Pure(), ConstantLike())
+
+    assembly_format = "$value attr-dict"
+
+    def __init__(self, value: bool):
+        value_attr = IntegerAttr(-1 if value else 0, 1)
+        super().__init__(properties={"value": value_attr}, result_types=[BoolType()])
+
+    @property
+    def value(self) -> bool:
+        return self.value_attr.value.data != 0
```

Comment:
?
```suggestion
        return bool(self.value_attr)
```
Deserves a test?

Reply:
It's tested in `test_smt.py`:
```
    op = ConstantBoolOp(True)
    assert op.value is True
    assert op.value_attr == IntegerAttr(-1, 1)
```

Reply:
I personally think that using these `__bool__` are a bad idea in Python, but I'll change it.